src/main/java/javacc/output/java/FilesGenerator.java

```diff
@@ -59,7 +59,7 @@ void initializeTemplateEngine() throws IOException {
         String filename = grammar.getFilename();
         File dir = new File(filename).getCanonicalFile().getParentFile();
         TemplateLoader templateLoader = new MultiTemplateLoader(new FileTemplateLoader(dir), 
-                                                                new ClassTemplateLoader(this.getClass(), ""));
+                                                                new ClassTemplateLoader(this.getClass(), "/templates/java"));

```

Comment:
To keep simply, the templates files from the src/main/resources/templates/java should have the same folder in jar (templates/java). Your ant keep src/main/resources/templates/java and copy those files in output/java (in the same package than FilesGenerator). If you want to keep this foolder (and not set templates.java in FilesGenerator, we should move src/main/resources/templates/java to src/main/resources/javacc/output/java to keep simply.

Reply:
Well, in principle, it hardly matters what folder any templates are in in the jar. Actually, it doesn't matter that much what folder they are in on one's hard disk for development, except that I find it a little bit less comfortable to have them buried so deep. But anyway, I put the templates in the same folder as the javacc.output.java classes because then you just use FreeMarker's ClassTemplateLoader with the empty string and it just finds the templates the same place it found the classfile, which seems quite straightforward to me. Actually, that is [here](https://github.com/javacc21/javacc21/blob/master/src/main/java/javacc/output/java/FilesGenerator.java#L61)

A funny thing about that is that I just noticed that the line after that, line 63, seems to be redundant. However, that must be because I originally had that line, but then decided it made sense to give templates priority if there are some in the same directory as the grammar, so I switched to MultiTemplateLoader but then probably never removed the line after that. I just commented it out and everything works as before!

Well, I guess, if you can keep everything working, it hardly matters. But my honest reaction is that there isn't a lot of value added in changing this, of all things... I mean, it works pretty straightforwardly and very few people are looking inside the .jar file anyway examining the directory layout in there. A few people, maybe like the people who buy a car and open the hood and start poking around at the engine (not me!) and so on, but that's a very small minority!

Speaking of layouts, though, what do you think of moving from javacc.* to com.javacc.*? (I don't think I'll go to io.javacc.* since I don't own the javacc.io domain and it's not worth paying 50 euros for it. Every year...)

Reply:
One of big reason why I would like to keep the same folder (from src/main/resources/templates) and inside JAR is that I can use javacc.Main by doing a Run/Debug Eclipse IDE and I'm sure that I use last version of javacc (last version of templates, java classes) and it avoid creating the javacc.jar when you change the ftl templates.

So if you wish to keep output/java, I think we should move those templates in src/main/resources/output/java. I find name is ot explicit, I think it's better to use templates/java, but for doing that, you need to do like I did.

Reply:
I'm a bit confused. If the templates are under src/main (independently of the exact subfolder) and, as part of the build culminating in the jar, you put everything needed under build/ then you are copying the files already anyway, so I'm having a hard time understanding where you gain something. Actually, using the /build directory as the classpath so you run/debug from the version on the file system is also what I do. But yes, it is true that this  requires the post-compile step of copying the templates over. In the Ant build, that is [here.](https://github.com/javacc21/javacc21/blob/master/build.xml#L68)

Oh, it occurs to me that it will load the templates in-place without copying in your local setup if you have src/main/resources/ on your run/test/debug classpath locally. Except yes then you need to load from /templates/java. So, okay, I guess I see the issue. 

Well... okay, if you want to put the templates in /templates/java for this reason, that is okay and yes, it is perhaps clearer. (Though, again, as regards the layout _inside_ the .jar file, we can be sure that just about nobody cares!) So... finally, okay, this seems okay. Do you want to change this as part of this PR or should I do it myself?

Reply:
>  But yes, it is true that this requires the post-compile step of copying the templates over. 

It's exactly the problem that I'm trying to explain you. Keep simple thing.

> Do you want to change this as part of this PR or should I do it myself?

I think a very clean fix is:

 * use only `/templates` path:

```java
... new ClassTemplateLoader(this.getClass(), "/templates"));
```

 * use `templateName = "java/ASTToken.java.ftl"` instead of using `templateName = "ASTToken.java.ftl"` because we must think about other language generation.